src/services/getEditsForFileRename.ts

```diff
@@ -44,7 +44,10 @@ namespace ts {
                 // If it resolved to something already, ignore.
                 if (checker.getSymbolAtLocation(importStringLiteral)) continue;
 
-                const resolved = program.getResolvedModuleWithFailedLookupLocationsFromCache(importStringLiteral.text, sourceFile.fileName);
+                if (!host.resolveModuleNameWithFailedLookupLocations && !host.fileExists) continue;
+                const resolved = host.resolveModuleNames
+                    ? host.resolveModuleNameWithFailedLookupLocations && host.resolveModuleNameWithFailedLookupLocations(importStringLiteral.text, sourceFile.fileName)

```

Comment:
Can you please make this getResolvedModuleWithFailedLookupLocations as well? We shouldnt be re-resolving modules but get the result from cache.

Reply:
But when we get those in `program.ts`, we aren't calling `resolveModuleNameWithFailedLookupLocations` but `resolveModuleNames`, so there are no failed lookup locations to cache. This will actually be the first time we call `resolveModuleNameWithFailedLookupLocations`, and we'll only call it for imports that didn't succeed.

Reply:
But its still import already resolved and failedlookups present in cache(eg in resolution cache).. You shouldn't have to resolve the module name again for this.

Reply:
I could try to modify `Project`'s implementation to look in the cache, would that suffice?

Reply:
Yes and also change the name of the method needed to ensure we arent asking for re-resolution but getting already resolved result.